### PR TITLE
pldm: Catch exceptions precisely and printing it

### PR DESCRIPTION
**Pull request by Patel-Kamalkumar**
Correcting catch blocks in PLDM repo to print all the
exceptions precisely so that pldm trace can be more
useful to debug the issues easily.

Trace sample:
Oct 20 02:14:22 p10bmc pldmd[742]: PLDM: Communication Error, rc =1

### CODE REVIEW

File: host-bmc/dbus_to_host_effecters.cpp [hunk 1]
```diff
@@ -190,7 +190,8 @@ void HostEffecterParser::processHostEffecterChangeNotification(
     }
     catch (const std::out_of_range& e)
     {
-        error("new state not found in json");
+        error("new state not found in json  ERROR={ERR_EXCEP}", "ERR_EXCEP",
```
Comment (jaypadath) [inline host-bmc/dbus_to_host_effecters.cpp:193]: Remove extra space after json

File: oem/ibm/libpldmresponder/oem_ibm_handler.cpp [hunk 2]
```diff
@@ -710,9 +710,9 @@ std::filesystem::path pldm::responder::oem_ibm_platform::Handler::getConfigDir()
         catch (const std::exception& e)
         {
             error(
-                "Error getting Names property , PATH={OBJ_PATH} Compatible interface = {INTF}",
+                "Error getting Names property , PATH={OBJ_PATH} Compatible interface = {INTF} ERROR={ERR_EXCEP}",
```
Comment (jaypadath) [inline oem/ibm/libpldmresponder/oem_ibm_handler.cpp:713]: Remove extra space after property

File: pldmd/pldmd.cpp [hunk 3]
```diff
@@ -135,7 +135,8 @@ static std::optional<Response>
             header.command = hdrFields.command;
             if (PLDM_SUCCESS != pack_pldm_header(&header, responseHdr))
             {
-                error("Failed adding response header");
+                error("Failed adding response header  ERROR={ERR_EXCEP}",
```
Comment (jaypadath) [inline pldmd/pldmd.cpp:138]: Remove extra space after header

File: pldmtool/pldm_bios_cmd.cpp [hunk 4]
```diff
@@ -714,14 +715,14 @@ class GetBIOSAttributeCurrentValueByHandle : public GetBIOSTableHandler
...
-            lg2::info("StringTable/AttrTable Unavaliable");
+            info("StringTable/AttrTable Unavaliable");
             return;
         }
 
         auto handle = findAttrHandleByName(attrName, *attrTable, *stringTable);
         if (!handle)
         {
-            lg2::error("Can not find the attribute {KEY0}", "KEY0", attrName);
+            error("Can not find the attribute {KEY0}", "KEY0", attrName);
```
Comment (jaypadath) [inline pldmtool/pldm_bios_cmd.cpp:725]: Change "Can not" to "Could not"

**Review by jaypadath — COMMENTED** [review]
Change commit message to "Correcting catch blocks in PLDM repo to print all the
exceptions precisely so that pldm trace can be more
useful to debug the issues easily."

Please add the Tested field also.

File: host-bmc/dbus_to_host_effecters.cpp (hunk 1 above)
Comment (Patel-Kamalkumar) [reply host-bmc/dbus_to_host_effecters.cpp]: done

File: pldmd/pldmd.cpp (hunk 3 above)
Comment (Patel-Kamalkumar) [reply pldmd/pldmd.cpp]: done

File: pldmtool/pldm_bios_cmd.cpp (hunk 4 above)
Comment (Patel-Kamalkumar) [reply pldmtool/pldm_bios_cmd.cpp]: done

**Review by jaypadath — APPROVED** [review]

**Pull request merged**